src/store/getters.ts

```diff
@@ -359,7 +691,201 @@ export const getShowCorpNum = (state: StateIF): 'colin' | 'mras' | false => {
+  return state.stateModel.newRequestModel.exitModalVisible
+}
+
+// JSON Request constructors
+export const getNrRequestNames = (state: StateIF): RequestNameI[] => {
+  const nameChoices = getNameChoices(state)
+  const nrNames = getNrNames(state)
+
+  const defaultValues = {
+    name_type_cd: getAssumedName(state) ? 'AS' : 'CO',
+    consent_words: '',
+    conflict1: '',
+    conflict1_num: ''
+  }
+
+  // Check to make sure there are nameChoices that have been set
+  // const nameChoicesAreSet = (parseNameChoices(nameChoices).length > 0)
+
+  let requestNames = []
+  if (nameChoices) {
+    // We only allow three choices
+    let choiceIdx = 1
+    while (choiceIdx <= 3) {
+      if (nameChoices[`name${choiceIdx}`] as boolean) {
+        let combinedName = nameChoices[`name${choiceIdx}`]
+        if (getEntityTypeCd(state) && designations[getEntityTypeCd(state)]?.end) {
+          let des = nameChoices[`designation${choiceIdx}`]
+          if (des && !combinedName.endsWith(des)) {
+            combinedName = combinedName + ' ' + des
+          }
+          requestNames.push({
+            name: combinedName,
+            designation: nameChoices[`designation${choiceIdx}`],
+            choice: choiceIdx,
+            ...defaultValues
+          })
+        } else {
+          requestNames.push({
+            name: combinedName,
+            designation: '',
+            choice: choiceIdx,
+            ...defaultValues
+          })
+        }
+      }
+      choiceIdx++
+    }
+  } else {
+    // Just use the 'name' property to fill in the requestName
+    if (getEntityTypeCd(state) && getLocation(state) === 'BC' && designations[getEntityTypeCd(state)]?.end) {
+      requestNames.push({
+        name: getName(state),
+        designation: this.splitNameDesignation.designation,
+        choice: 1,
+        ...defaultValues
+      })
+    } else {
+      requestNames.push({
+        name: getName(state),
+        designation: '',
+        choice: 1,
+        ...defaultValues
+      })
+    }
+  }
+  requestNames = requestNames.map((requestName, idx) => {
+    if (nrNames) {
+      const existingName = nrNames.find(nrName => nrName.choice === requestName.choice)
+      if (existingName) {
+        return {
+          ...existingName,
+          // Merge in requestName form values
+          ...requestName,
+          // Merge conflicts and consent words
+          consent_words: !existingName.consent_words
+            ? this.consentWords.length > 0 ? this.consentWords : ''
+            : existingName.consent_words,
+          conflict1: !existingName.conflict1
+            ? this.consentConflicts.name
+            : existingName.conflict1,
+          conflict1_num: existingName.conflict1_num ? existingName.conflict1_num : ''
+        } as RequestNameI
+      }
+    }
+
+    return { ...requestName } as RequestNameI
+  })
+
+  return requestNames
+}
+
+export const getDraftNameReservation = (state: StateIF): DraftReqI => {
```

Comment:
I think this should be out of the store and in a mixin or something. It can always use the getters to get its data. I'm nervous about it using `state`.... who knows if state is updated when this code runs?